Top-Level Requirements.md

```diff
@@ -1,63 +1,86 @@
+    * 6.2.2 Scan for fiducial markers in order to localize the robot.
+* 6.3 Once the subsystem is localized the localization subsystem **shall** signal the executive system.
+### 7.  Navigation Subsystem
+#### 7.1  Fundamental
+* 7.1.1 The navigation subsystem **shall** define a goal as a position and quaternion orientation.
+* 7.1.2 The navigation subsystem **shall** provide and read necessary and sufficient data to calculate path information.
+* 7.1.3 The navigation subsystem **shall** move the rover towards the goal, based on path information.
+* 7.1.4 The navigation subsystem **shall** cease operation upon arriving at a goal within a set tolerance.
+* 7.1.5 The navigation subsystem **shall**  provide appropriate error handling procedures.
+#### 7.2  Pre-Mining
+* 7.2.1 The software **shall** define a point at a configurable distance directly East of the bin , and within the mining zone as the initial goal.
+* 7.2.2 On navigating to the bin the subsystem **shall** optimize the current goal to minimize driving time for the rover.
+* 7.2.3 The algorithm **shall** define a line segment that represents the set of “minable” locations using the points L0, L1, and L2, where:
+    * 7.2.3.1 The point L1 is defined as a point directly East of the bin X meters, where X is the contest requirement for mining distance plus some configurable safety threshold.
+    * 7.2.3.2 The point L0 is defined as a point directly East of the robot and directly North or South of L1.
+    * 7.2.3.3 The point L2 is defined as a point flipped from L0 over L1.
+* 7.2.3 The software **shall** have a configurable time value GOAL_RECALC_DELTA that represents the time between recalculations
+* 7.2.4 The software **shall** calculate the nearest point on the the Mining Line every GOAL_RECALC_DELTA time units and use that nearest point as the navigation goal
+#### 7.3  Post-Mining
+* 7.3.1 The software **shall** define a static point at a configurable distance directly East of the bin as the goal and navigate there.
+### 8.  Mining Subsystem
+* 8.1 The mining subsystem **shall** be provided a time limit and not operate longer than that time limit.
+* 8.2 The mining subsystem **shall** hold a predefined mining procedure.
+* 8.3 The mining subsystem **shall** follow a predefined series of mining movements during the mining procedure.
+* 8.4 The mining procedure **shall** define an ordered set of mining locations (and dumping locations) relative to the robot position
+* 8.5 The mining procedure **shall** further be divided into atomically-operating subprocedures (scoops)
+* 8.6 The mining procedure **shall** always check timing flags before beginning scoops
+* 8.7 The mining procedure **shall** be terminated upon detecting timing flags
+* 8.8 Upon completing the mining procedure, the mining subsystem **shall**:
+    * 8.8.1 Define an “obstacle” representing the digging hole and send it to the navigation subsystem.
+    * 8.8.2 Return the mining arm to the starting position.
```

Comment:
8.9 The mining subsystem **shall** provide a method to move the digging arm out of the bin to a predefined dumping position to enable bin movement without interference

Reply:
(this is my proposed addition here, feel free to change it to a way that's better phrased)

Reply:
Upon further investigation of the system architecture, there are two ways to do this: the first is as I proposed initially, where the request to move the arm goes through the digging action server. The other option is that we can actually remove this extra requirement and have the dumping system interact with the Arm Action Server directly to move the arm, removing the middleman.

The only argument against this alternative option is that it involves bypassing another system's control of this action, but I'm not too concerned about this as digging and dumping should never be operating at the same time, so I'd propose actually not including this requirement and just letting the dumping system interface with the arm action server directly.

I'll leave the original addition here @RyanBerge just for reference, but please note that my updated proposal is to actually not include it.